moto/core/utils.py

```diff
@@ -411,6 +411,7 @@ def extract_region_from_aws_authorization(string):
 class BackendDict(dict):
     def __init__(self, fn, service_name):
         self.fn = fn
+        self.service_name = service_name
```

Comment:
What's the reason behind this change?

Reply:
seems this one slipped in. we're patching the backend dict to return account specific instances, to be able to do that we need to remember which service the originally patched backend belongs to. it's basically just to help extensibility.